builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,598 @@
+		Resources:     resources,
+		Health:        jobResource.Health,
+		HealthMessage: jobResource.HealthMessage,
+	}
+
+	// update output based on main health state
+	step.Update("Finished building report for Nomad platform")
+	step.Done()
+
+	// NOTE(briancain): Replace ui.Status with StepGroups once this bug
+	// has been fixed: https://github.com/hashicorp/waypoint/issues/1536
+	st := ui.Status()
+	defer st.Close()
+
+	// More UI detail for non-ready resources
+	for _, resource := range result.Resources {
+		if resource.Health != sdk.StatusReport_READY {
+			st.Step(terminal.StatusWarn, fmt.Sprintf("Resource %q is reporting %q", resource.Name, resource.Health.String()))
+		}
+	}
+
+	return &result, nil
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// List of task group names which are to be promoted
+	Groups []string `hcl:"groups,optional"`
+
+	// If true, fails the canary deployment
+	FailDeployment bool `hcl:"fail_deployment,optional"`
+}
+
+type nomadClient struct {
+	NomadClient *api.Client
+}
+
+func (r *Releaser) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(docs.FromConfig(&ReleaserConfig{}))
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Promotes a Nomad canary deployment initiated by a Nomad jobspec deployment.
+
+If your Nomad deployment is configured to use canaries, this releaser plugin lets
+you promote (or fail) the canary deployment. You may also target specific task
+groups within your job for promotion, if you have multiple task groups in your canary
+deployment.
+
+-> **Note:** Using the ` + "`-prune=false`" + ` flag is recommended for this releaser. By default,
+Waypoint prunes and destroys all unreleased deployments and keeps only one previous
+deployment. Therefore, if ` + "`-prune=false`" + ` is not set, Waypoint may delete
+your job via "pruning" a previous version. See [deployment pruning](/docs/lifecycle/release#deployment-pruning)
+for more information.
+
+### Release URL
+
+If you want the URL of the release of your deployment to be published in Waypoint,
+you must set the meta 'waypoint.hashicorp.com/release_url' in your jobspec. The
+value specified in this meta field will be published as the release URL for your
+application. In the future, this may source from Consul.
+
+`)
+
+	doc.Example(`
+// The waypoint.hcl file
+release {
+	use "nomad-jobspec-canary" {
+		groups = [
+			"app"
+		]
+		fail_deployment = false
+	}
+}
+
+// The app.nomad.tpl file
+job "web" {
+  datacenters = ["dc1"]
+
+  group "app" {
+		network {
+      mode = "bridge"
+      port "http" {
+        to = 80
+      }
+		}
+
+		// Setting a canary in the update stanza indicates a canary deployment
+		update {
+			max_parallel = 1
+			canary       = 1
+			auto_revert  = true
+			auto_promote = false
+			health_check = "task_states"
+		}
+
+		service {
+			name = "app"
+			port = 80
+			connect {
+				sidecar_service {}
+			}
+		}
```

Comment:
This job block needs some formatting help. I'm guessing it's just a mix of tabs and whitespace! `network {` on line 490 looks a bit off too.

Reply:
You're absolutely right, it's possibly something with the linter in one of the linter plugins I'm using in my IDE (Atom) confusing Go with markdown 😅 should be fixed up now, with the help of good old vim.